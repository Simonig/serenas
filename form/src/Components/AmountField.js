import React, {Component} from 'react';

export default class AmountField extends Component {

	constructor() {
		super();
		this.formatAmount = this.formatAmount.bind(this)
	}

	formatAmount(e) {
		const {value, name} = e.target;
		const payload = {value, name};

		payload.value = Math.ceil(payload.value / 100) * 100;
		if (payload.value > 25000) payload.value = 25000;
		if (payload.value < 1000) payload.value = 1000;
		this.props.onChange(payload)

	}

	render() {
		const {value, name, label, onChange} = this.props;

		return (

			<div className="form-group">

				<label>Amount</label>

				<input
					type="number"
					id="amount"
					placeholder={label}
					value={value}
					className="form-control"
					step={100}
					name={name}
					onBlur={this.formatAmount}
					onChange={(e) => onChange(e.target)}
				/>
			</div>


		)
	}
}
