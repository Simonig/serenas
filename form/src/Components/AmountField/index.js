import React, { Component } from 'react';
import PropTypes from 'prop-types';

class AmountField extends Component {
  formatAmount = (e) => {
    const { value, name } = e.target;
    const payload = { value, name };

    payload.value = Math.ceil(payload.value / 100) * 100;
    if (payload.value > 25000) payload.value = 25000;
    if (payload.value < 1000) payload.value = 1000;

    this.props.onChange(payload);
  };

  render() {
    const {
      value, name, label, onChange,
    } = this.props;

    return (
      <div className="form-group">

        <label>Amount</label>

        <input
          type="number"
          id="amount"
          placeholder={label}
          value={value}
          className="form-control"
          step={100}
          name={name}
          onBlur={this.formatAmount}
          onChange={(e) => onChange(e.target)}
        />
      </div>
    );
  }
}

AmountField.defaultProps = {
  onChange: () => {},
  value: 1000,
  label: '',
};

AmountField.propTypes = {
  onChange: PropTypes.func,
  value: PropTypes.number,
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
};

export default AmountField;
