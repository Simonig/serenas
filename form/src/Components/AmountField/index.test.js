import React from 'react';
import { shallow } from 'enzyme';
import AmountField from './index';

describe('AmountField Render', () => {
  it('should render with defaults', () => {
    const tree = shallow(<AmountField name="test" />);
    expect(tree).toMatchSnapshot();
  });

  it('should render an input with right attributes', () => {
    const tree = shallow(<AmountField name="test" label="testLabel" value={1100} />);
    const inputs = tree.find('input');

    const input = inputs.first();

    expect(inputs.length).toEqual(1);
    expect(input.prop('name')).toEqual('test');
    expect(input.prop('placeholder')).toEqual('testLabel');
    expect(input.prop('value')).toEqual(1100);
  });
});

describe('AmountField Functionality', () => {
  it('should step the number on blur', () => {
    const onChange = jest.fn();
    const tree = shallow(<AmountField name="test" value={1001} onChange={onChange} />);
    const input = tree.find('input').first();

    input.simulate('blur', {
      target: {
        value: input.prop('value'),
        name: input.prop('name'),
      },
    });

    expect(onChange).toHaveBeenCalledWith({
      name: 'test',
      value: 1100,
    });
  });

  it('should limit the max number on blur', () => {
    const onChange = jest.fn();
    const tree = shallow(<AmountField name="test" value={99999} onChange={onChange} />);
    const input = tree.find('input').first();

    input.simulate('blur', {
      target: {
        value: input.prop('value'),
        name: input.prop('name'),
      },
    });

    expect(onChange).toHaveBeenCalledWith({
      name: 'test',
      value: 25000,
    });
  });

  it('should limit the min number on blur', () => {
    const onChange = jest.fn();
    const tree = shallow(<AmountField name="test" value={-99999} onChange={onChange} />);
    const input = tree.find('input').first();

    input.simulate('blur', {
      target: {
        value: input.prop('value'),
        name: input.prop('name'),
      },
    });

    expect(onChange).toHaveBeenCalledWith({
      name: 'test',
      value: 1000,
    });
  });
});
