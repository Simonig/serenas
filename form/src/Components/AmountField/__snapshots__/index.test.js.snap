// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountField Render should render with defaults 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AmountField
    label=""
    name="test"
    onChange={[Function]}
    value={1000}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          Amount
</label>,
        <input
          className="form-control"
          id="amount"
          name="test"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          step={100}
          type="number"
          value={1000}
/>,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Amount",
        },
        "ref": null,
        "rendered": "Amount",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control",
          "id": "amount",
          "name": "test",
          "onBlur": [Function],
          "onChange": [Function],
          "placeholder": "",
          "step": 100,
          "type": "number",
          "value": 1000,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            Amount
</label>,
          <input
            className="form-control"
            id="amount"
            name="test"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            step={100}
            type="number"
            value={1000}
/>,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Amount",
          },
          "ref": null,
          "rendered": "Amount",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "id": "amount",
            "name": "test",
            "onBlur": [Function],
            "onChange": [Function],
            "placeholder": "",
            "step": 100,
            "type": "number",
            "value": 1000,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
